salalib/ntfp.cpp

```diff
@@ -73,15 +76,15 @@ void NtfMap::fitBounds(const Line& li)
    }
 }
 
-void NtfMap::addGeom(int layer, NtfGeometry& geom)
+void NtfMap::addGeom(NtfLayer& layer, NtfGeometry& geom)

```

Comment:
The only internal state this function changes with the new implementation is `m_line_count` - I am not sure that is appropriate if the layer itself gets passed in - the function is protected, so a derived class could pass in a completely unrelated layer and the internal state of the class would be all over the shop.
I think this function needs to either work on class data, or be a completely separate helper that is not a class member (and e.g. the count encountered here so call would be `m_line_count += addGeom(layer, geom);`)